Doc/whatsnew/3.8.rst

```diff
@@ -242,6 +242,11 @@ Optimizations
   Note that this means that instances can no longer be weak-referenced and
   that arbitrary attributes can no longer be added to them.
 
+* The :class:`list` constructor does not overallocate the internal item buffer
+  if the input iterable has a known size (the input implements ``__len__``).
+  This makes the created list 12% smaller on average and it speeds the creation
+  of very large lists. (Contributed by Pablo Galindo in :issue:`33234`.)
```

Comment:
"and it speeds the creation of very large lists" even if it's true, I'm not sure that it's worth it to mention it. The most important part is the memory footprint, no?

Reply:
Yup, I will remove that sentence.

Reply:
Done in 764858a